summarize/src/aggregate.rs

```diff
@@ -0,0 +1,556 @@
+            if let ExtremaSources::One(description) = descriptions {
+                match (description.start, description.end) {
+                    (SamplePoint::Start(start), SamplePoint::End(end)) => {
+                        assert_eq!(start, end);
+                        start.this.to_string()
+                    }
+
+                    (SamplePoint::Start(outer), SamplePoint::Start(inner))
+                    | (SamplePoint::Start(outer), SamplePoint::Instant(inner)) => {
+                        assert_eq!(inner.parent.as_ref(), Some(&outer.this));
+                        format!("in {}, before {}", outer.this, inner.this)
+                    }
+
+                    (SamplePoint::End(inner), SamplePoint::End(outer))
+                    | (SamplePoint::Instant(inner), SamplePoint::End(outer)) => {
+                        assert_eq!(inner.parent.as_ref(), Some(&outer.this));
+                        format!("in {}, after {}", outer.this, inner.this)
+                    }
+
+                    (SamplePoint::End(first), SamplePoint::Start(second))
+                    | (SamplePoint::Instant(first), SamplePoint::Start(second))
+                    | (SamplePoint::End(first), SamplePoint::Instant(second))
+                    | (SamplePoint::Instant(first), SamplePoint::Instant(second)) => {
+                        assert_eq!(first.parent, second.parent);
+                        if let Some(parent) = &first.parent {
+                            format!(
+                                "in {},\n    between {}\n        and {}\n",
+                                parent, first.this, second.this
+                            )
+                        } else {
+                            format!("between {} and {}", first.this, second.this)
+                        }
+                    }
+                }
+            } else {
+                let count = descriptions.count();
+                format!(
+                    "{} occurrences, or {:.2}%",
+                    count,
+                    (count as f64) / (intervals_count as f64) * 100.0
+                )
+            }
+        };
+
+    println!("Smallest {} durations:", durations.smallest.len());
+    for (duration, descriptions) in durations.smallest {
+        println!("  {} ns: {}", duration.as_nanos(), describe(descriptions));
+    }
+    println!("");
+    println!("Largest {} durations:", durations.largest.len());
+    for (duration, descriptions) in durations.largest {
+        println!("  {} ns: {}", duration.as_nanos(), describe(descriptions));
+    }
+    println!("");
+    println!("Smallest {} variances:", variances.smallest.len());
+    for (variance, descriptions) in variances.smallest {
+        println!(
+            "  ±{} ns: {}",
+            variance.range_size.as_nanos() as f64 / 2.0,
+            describe(descriptions)
+        );
+    }
+    println!("");
```

Comment:
Nit: you can just say

```suggestion
    println!();
```